middleware/ayncHandler.js

```diff
@@ -0,0 +1,34 @@
+import { Prisma } from "@prisma/client";
+
+
+/*  // 사용하기는 간단하나, 잘못된 요청에도 500 오류가 발생되는 듯
+const asyncHandler = (fn) => (req, res, next) => {
+  Promise.resolve(fn(req, res, next)).catch(next);
+};
+*/
+function asyncHandler(handler) {
+  return async function (req, res, next) {
+    try {
+      await handler(req, res);
+    } catch (e) {
+      if (res.headersSent) {
+        // 이미 응답이 전송된 경우에는 next로 에러를 전달하여 Express가 기본 처리를 하도록 함
+        return next(e); 
+      }      
+      if (
+        e.name === 'StructError' ||
+        e instanceof Prisma.PrismaClientValidationError
+      ) {
+        res.status(400).send({ message: e.message });
+      } else if (
+        e instanceof Prisma.PrismaClientKnownRequestError &&
+        e.code === 'P2025'
```

Comment:
@BuffSo `P2025`는 무슨 에러코드인가요?

Reply:
Prisma에서 발생하는 오류인지 확인하고 
그 중 에러 코드가  'P2025'인 경우 `해당 레코드를 찾을 수 없는 경우` 에 맞게 에러 메시지를 처리하는 코드입니다.
강의 시간에 배운대로 적용했습니다 ^^;;

P2025
"An operation failed because it depends on one or more records that were required but not found. {cause}"

https://www.prisma.io/docs/orm/reference/error-reference